app/src/androidTest/java/com/sdp/swiftwallet/uiTest/TransactionActivityTest.java

```diff
@@ -0,0 +1,200 @@
+package com.sdp.swiftwallet.uiTest;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+
+import android.content.Intent;
+
+import androidx.test.core.app.ActivityScenario;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import com.sdp.cryptowalletapp.R;
+import com.sdp.swiftwallet.SwiftWalletApp;
+import com.sdp.swiftwallet.di.TransactionHistoryProducerModule;
+import com.sdp.swiftwallet.domain.model.Currency;
+import com.sdp.swiftwallet.domain.model.Transaction;
+import com.sdp.swiftwallet.domain.repository.TransactionHistoryProducer;
+import com.sdp.swiftwallet.domain.repository.TransactionHistorySubscriber;
+import com.sdp.swiftwallet.presentation.transactions.TransactionActivity;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.runner.RunWith;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import dagger.Module;
+import dagger.Provides;
+import dagger.hilt.InstallIn;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
+import dagger.hilt.components.SingletonComponent;
+
+@UninstallModules(TransactionHistoryProducerModule.class)
+@HiltAndroidTest
+@RunWith(AndroidJUnit4.class)
+public class TransactionActivityTest {
+
+    @Module
```

Comment:
Oh you did have time to fix it! nice work

Reply:
Yep, I'm just not sure why the tests aren't passing on the CI still